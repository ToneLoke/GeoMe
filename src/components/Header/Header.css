.trigger{
  padding: 0 16px;
  cursor: pointer;
  transition: color .3s;
  font-size: 25px;
}
.trigger:hover {
  color: RGBA(16, 90, 145, 1);
}
.logocity {
  letter-spacing: 5px;
  font-size: 25px;
  color: RGBA(16, 50, 105, 1);
  font-weight: 900;
}
