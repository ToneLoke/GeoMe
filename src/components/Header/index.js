import React from 'react'
import { Layout, Icon } from 'antd'
const { Header } = Layout

const TopHeader = ({city, country}) => (
  <Header className='title'>
    <Icon
      className='trigger'
      type={'cloud-o'}
    />
    GeoMe
    <Icon
      className='trigger'
      type={'smile'}
    />
    <span id='geo-city'>{city}</span>, <span id='geo-country'>{country}</span>
  </Header>
)

export default TopHeader
