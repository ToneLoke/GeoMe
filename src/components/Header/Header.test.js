import React from 'react'
import ReactDOM from 'react-dom'
import Header from './index'
import { mount } from 'enzyme'
import renderer from 'react-test-renderer'

test('top header renders without crashing', () => {
  const div = document.createElement('div')
  ReactDOM.render(<Header />, div)
})

test('to header renders with the correct city name', () => {
  const city = 'Denver'
  const country = 'US'
  const rendered = renderer.create(
    <Header city={city} country={country} />
  )
  expect(rendered.toJSON()).toMatchSnapshot()
})
