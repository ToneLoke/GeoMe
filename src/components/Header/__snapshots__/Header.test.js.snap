// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`to header renders with the correct city name 1`] = `
<div
  className="title ant-layout-header"
>
  <i
    className="anticon anticon-cloud-o trigger"
  />
  GeoMe
  <i
    className="anticon anticon-smile trigger"
  />
  <span
    id="geo-city"
  >
    Denver
  </span>
  , 
  <span
    id="geo-country"
  >
    US
  </span>
</div>
`;
